src/devices/Button/GpioButton.cs

```diff
@@ -13,56 +13,61 @@ namespace Iot.Device.Button
+            : this(buttonPin, TimeSpan.FromTicks(DefaultDoublePressTicks), TimeSpan.FromMilliseconds(DefaultHoldingMilliseconds), isPullUp, hasExternalResistor, gpio, shouldDispose, debounceTime)
         {
         }
 
         /// <summary>
         /// Initialization of the button.
         /// </summary>
         /// <param name="buttonPin">GPIO pin of the button.</param>
-        /// <param name="pinMode">Pin mode of the system.</param>
         /// <param name="doublePress">Max ticks between button presses to count as doublepress.</param>
         /// <param name="holding">Min ms a button is pressed to count as holding.</param>
+        /// <param name="isPullUp">True if the Gpio is either pulled up in hardware or in the Gpio configuration (see <paramref name="hasExternalResistor"/>. False if instead the Gpio is pulled down.</param>
+        /// <param name="hasExternalResistor">When False the pull resistor is configured using the Gpio PinMode.InputPullUp or PinMode.InputPullDown (if supported by the board). Otherwise the Gpio is configured as PinMode.Input.</param>
         /// <param name="gpio">Gpio Controller.</param>
         /// <param name="shouldDispose">True to dispose the GpioController.</param>
         /// <param name="debounceTime">The amount of time during which the transitions are ignored, or zero</param>
-        public GpioButton(int buttonPin, TimeSpan doublePress, TimeSpan holding, GpioController? gpio = null, bool shouldDispose = true, PinMode pinMode = PinMode.InputPullUp, TimeSpan debounceTime = default(TimeSpan))
+        public GpioButton(int buttonPin, TimeSpan doublePress, TimeSpan holding, bool isPullUp = true, bool hasExternalResistor = false,
+            GpioController? gpio = null, bool shouldDispose = true, TimeSpan debounceTime = default)
             : base(doublePress, holding, debounceTime)
         {
             _gpioController = gpio ?? new GpioController();
             _shouldDispose = shouldDispose;
             _buttonPin = buttonPin;
-            _pinMode = pinMode;
-            _debounceTime = debounceTime;
+            HasExternalResistor = hasExternalResistor;
 
-            if (_pinMode == PinMode.Input | _pinMode == PinMode.InputPullDown | _pinMode == PinMode.InputPullUp)
-            {
-                _gpioController.OpenPin(_buttonPin, _pinMode);
-                _gpioController.RegisterCallbackForPinValueChangedEvent(_buttonPin, PinEventTypes.Falling | PinEventTypes.Rising, PinStateChanged);
-            }
-            else
-            {
-                throw new ArgumentException("GPIO pin can only be set to input, not to output.");
-            }
+            _eventPinMode = isPullUp ? PinMode.InputPullUp : PinMode.InputPullDown;
+            _gpioPinMode = hasExternalResistor
+                ? _gpioPinMode = PinMode.Input
+                : _gpioPinMode = _eventPinMode;
+
+            _gpioController.OpenPin(_buttonPin, _gpioPinMode);
```

Comment:
This may need some error handling here, since OpenPin can fail (invalid pin number, unsupported pin mode, etc.). In this case, the gpioController needs to be disposed immediately (if _shouldDispose is true), since we're in a constructor. The caller has no way of using a finaly arount the exception.